File: languages/go/entity/identityV2/issuer_state_receipt.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type IssuerStateReceipt struct {
	TxHash string
}

func NewIssuerStateReceiptFromProto(s *proto.IssuerStateReceipt) IssuerStateReceipt {
	if s == nil {
		return IssuerStateReceipt{}
	}
	return IssuerStateReceipt{
		TxHash: s.TxHash,
	}
}

func (i IssuerStateReceipt) ToProto() *proto.IssuerStateReceipt {
	return &proto.IssuerStateReceipt{
		TxHash: i.TxHash,
	}
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/IssuerStateReceipt.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a receipt for the issuer's state.
 */
public class IssuerStateReceipt {
  private final String txHash;

  /**
   * Constructs an IssuerStateReceipt object with the specified parameters.
   * 
   * @param txHash
   */
  public IssuerStateReceipt(String txHash) {
    this.txHash = txHash;
  }

  public static IssuerStateReceipt fromProto(BloockIdentityEntities.IssuerStateReceipt res) {
    return new IssuerStateReceipt(res.getTxHash());
  }

  public BloockIdentityEntities.IssuerStateReceipt toProto() {
    return BloockIdentityEntities.IssuerStateReceipt.newBuilder().setTxHash(this.txHash).build();
  }

  /**
   * Gets the transaction hash of the issuer state receipt.
   * 
   * @return
   */
  public String getTxHash() {
    return txHash;
  }
}
